ttnn/cpp/ttnn/tensor/tensor_utils.hpp

```diff
@@ -46,17 +46,20 @@ static std::size_t compute_volume(const T& shape) {
     return volume;
 }
 
-static std::vector<std::size_t> compute_strides(Shape shape) {
+static std::vector<uint32_t> compute_strides(Shape shape) {
     auto num_elements = compute_volume(shape);
-    std::vector<std::size_t> strides;
+    std::vector<uint32_t> strides;
     for (std::int32_t index = 0; index < shape.rank(); index++) {
+        if(shape[index] == 0) {
+            TT_THROW("Shape can't have dimension {} as zero {}", index, shape);
+        }
```

Comment:
Just saw this condition. See my other comment: https://github.com/tenstorrent/tt-metal/pull/12286/files#r1746127562